x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/Transform.java

```diff
@@ -342,14 +350,62 @@ public void close() {
     public void cleanUpFeature(
         ClusterService clusterService,
         Client client,
-        ActionListener<ResetFeatureStateResponse.ResetFeatureStateStatus> listener
+        ActionListener<ResetFeatureStateResponse.ResetFeatureStateStatus> finalListener
     ) {
+
+        ActionListener<ResetFeatureStateResponse.ResetFeatureStateStatus> unsetResetModeListener = ActionListener.wrap(
+            success -> client.execute(SetResetModeAction.INSTANCE, SetResetModeActionRequest.disabled(), ActionListener.wrap(
+                resetSuccess -> finalListener.onResponse(success),
+                resetFailure -> {
+                    logger.error("failed to disable reset mode after otherwise successful transform reset", resetFailure);
+                    finalListener.onFailure(
+                        ExceptionsHelper.serverError(
```

Comment:
I am not sure about this one. It lacks a proper status code. If I get it right, it does not set one, which means it is based on the one from `resetFailure`, which could be anything. I think a status exception with a proper code would be better. My gut feeling would be a `503` - but that depends on the failure. Related to my comment above: what should a user do? A `503` would indicate a retry.

Reply:
`serverError` should throw an internal server error. If it doesn't I am misusing it and should throw an internal server error.